docs/decisions/0003-add-access-policy-functionality.rst

```diff
@@ -0,0 +1,313 @@
+     if self.access_method == AccessMethods.DIRECT:
+        return self.subsidy_client.redeem(learner_id, content_key)
+     if self.access_method == AccessMethods.REQUEST:
+        return self.subsidy_client..request_redemption(learner_id, content_key)
+   return None
+
+get redemptions
+---------------
+Given a content key and learner identifier,
+answers what redemption records, if any, exist for the learner and content via this policy.
+
+::
+
+   if self.access_method == AccessMethods.DIRECT:
+     return self.subsidy.has_redeemed(learner_id, content_key)  # pylint: disable=no-member
+   elif self.access_method == AccessMethods.REQUEST:
+     return self.subsidy.has_requested(learner_id, content_key)
+   else:
+     raise ValueError(f"unknown access method {self.access_method}")
+
+Use custom single-table inheritance pattern
+-------------------------------------------
+
+Our most common query patterns will require that we find all applicable policies for a given enterprise - there
+might be several at a given time. As described in the **Rejected Alternatives** section below, the base Django
+patterns for abstract or multi-table inheritance have some drawbacks for this query pattern that would ultimately
+require us to perform multiple database selects to get the policy records we want
+*that are wrapped by the Django model child class we want*.  Luckily, Django provides one of the key building blocks
+of `single-table inheritance`_ in the form of `proxy models`_.  Single-table inheritance is a means by which a field
+in the database table identifies which class (Django model) in the object hierarchy the record belongs to - all the fields
+of all the classes in the heirarchy are stored in the same table, hence "single table" inheritance.  A good
+`roadmap for single-table inheritance in Django`_ exists and will be the basis of our custom implementation of this pattern.
+
+Our custom implementation of this pattern consists of:
+
+- Concrete parent model with all fields, including all the fields for all of the children
+- A type/Discriminator field of some kind will also be required on this model.
+- The child policy models (as described in **Context** above) become proxy model sub-classes
+  to encapsulate their specific business logic.
+- A parent class query manager (eg ``PolicyManager``) which queries the main table and
+  "type-casts" the individual response records into the correct Proxy Model instance. This allows us to
+  query one table to find all policies for a given customer, catalog, etc. It also lets us query one table
+  for a specific policy record, by the UUID primary key.
+- The Django ORM requires us to add a small wrapper that takes the ORM results,
+  which will be ``SubsidyAccessPolicy`` records (i.e. the base model), and return instead a new collection
+  with each item properly typed as the appropriate child/proxy model.
+
+As a consequence, this may mean we don’t interact via queryset in some cases, for example:
```

Comment:
To me it makes more sense to say "As a consequence, this avoids the need to write complex branching logic and/or complex querysets to cover multiple policy subtypes, for example:"